src/base/api.lua

```diff
@@ -265,7 +265,11 @@
 
 		if type(field.allowed) == "table" then
 			for i, item in ipairs(field.allowed) do
-				field.allowed[item:lower()] = item
+				if(type(item) == "function") then
+					field.allowed["dynamicvalidator"] = item
+				else
```

Comment:
I don't think this is required at all, it also doesn't support multiple functions in the allowed field. Please just change it to this as this block is just about adding lowercase variants to the list.
```suggestion
				if type(item) == "string" then
```

Reply:
I think there might be a bit of a misunderstanding here. The block here is just adding lowercase versions of items so that you can have "Item" and "item" but when you put "item" you get "Item". There's no lowercase version of a function, so doing anything for a function is wrong. Furthermore, I'm pretty sure if you called `dotnetsdk "dynamicvalidator"` it will accept it and probably error when calling the `explode` function on `dotnetsdk` in `vs2005_dotnetbase.lua` because `explode` probably expects a string.

So to reiterate my comment above, this lowercase thing should only happen when `type(item) == "string"` and nothing should happen for any other type.